Controller/Customer/DeleteCreditCard.php

```diff
@@ -73,8 +83,11 @@ public function execute()
             $id = $this->getRequest()->getParam('id');
             if($id){
                 $creditCard = $this->customerCreditCardFactory->create()->load($id);
-
-                if($creditCard && $creditCard->getId()){
+                if (
+                    $creditCard
+                    && $creditCard->getId()
+                    && $this->customerSession->getCustomerId() == $creditCard->getCustomerId()
```

Comment:
FYI: Usually Magento check session like `$this->customerSession->getCustomerId()` and also through the function in the `Customer\Session::isLoggedIn()` - it is also check the `customerId` and session for customer.